cmd/kubeseal/main.go

```diff
@@ -142,15 +143,41 @@ func prettyEncoder(codecs runtimeserializer.CodecFactory, mediaType string, gv r
 	return enc, nil
 }
 
-func openCertFile(certFile string) (io.ReadCloser, error) {
-	f, err := os.Open(certFile)
+func isFilename(name string) (bool, error) {
+	u, err := url.Parse(name)
 	if err != nil {
+		return false, err
+	}
+	return u.Scheme == "", nil
+}
+
+func openCertLocal(certURL string) (io.ReadCloser, error) {
```

Comment:
IMO we should rename it to `openCertURI()`

Reply:
except a local filename is not an URI.

We even have a test that highlights that: we check for a file called `"/?this/file/cannot/possibly/exist/right?"`; if we treat arguments to `--cert` as URIs, possibly implicitly in the `file://` scheme, we'd require users to URI-encode the paths.

I thus opted for "Local" meaning "fetch the file from the POV of the client", as opposed from "use the k8s api server proxy to fetch things from within the cluster"

Reply:
naming is hard; added comments

Reply:
> except a local filename is not an URI.

Looks like it is: https://tools.ietf.org/html/rfc8089

Reply:
> naming is hard; added comments

Fair, comments "fulfill" it, but I'd still raise rfc8089 to claim the openCertURI() rename :)

Reply:
sure, rfc8089 is invoked when the user literally provides a parameter that starts with `file:...`

It's incorrect to assume that any host pathname that is valid to open with `os.Open` can be prefixed with `file:` and transformed to a rf8090 uri though.

I'll just make a couple of examples, using `curl` but the Go's FileTransport behaves in the same way.

1. 

```bash
$ echo foo >'/tmp/cert%E3%84%93.pem'
$ echo bar >'/tmp/certㄓ.pem'
$ cat '/tmp/cert%E3%84%93.pem'
foo
$ curl 'file:///tmp/cert%E3%84%93.pem'
**bar**
```

2.

```bash
$ echo foo >'/tmp/cert.pem'
$ echo bar >'/tmp/cert.pem?'
$ cat  '/tmp/cert.pem'
foo
$ curl 'file:///tmp/cert.pem'
foo
$ cat '/tmp/cert.pem?'
bar
$ curl 'file:///tmp/cert.pem?'
**foo**
```

I'll spare you the examples with windows backslashes.

Reply:
that said, I hear you; the naming is confusing. So I split it further and made it explicit what is a filename what is an URI